PUBLISH.md

```diff
@@ -0,0 +1,21 @@
+# Instructions for publishing a new SDK version
+
+## Go
+
+1. Tag a `main` commit with `go/vX.Y.X`. This is enough for the new version to be considered released.
+2. Additionally, we can create a new release directly on GitHub. make sure to choose the correct tag, target, and
+   previous version/tag. That allows you to use the `Generate release notes` button to get a good starting point of the
+   release notes.
+
+## TypeScript
+
```

Comment:
TS/NPM released also need the `main` branch to be tagged right?

Reply:
Tagging `main` sounds like a good idea but it's not required - if you look at [SDK v1.2.0](https://www.npmjs.com/package/@chainlink/data-streams-sdk/v/1.2.0), you can see it on NPM but there is no `main` tag for it.

If we decide to change the process and make that tag a requirement, then this doc will have to change but that's a separate conversation. My only goal with this PR is to describe the current release process, so people don't need to rely on word of mouth in order to get unblocked.